mantidproject/mantid
# User feedback fixes to D7YIGPositionCalibration

This PR fixes the issues and new feature requests brought up by D7 users:

- [x] Unwanted renaming of the workspace provided through InputWorkspace property
- [x] Include Chi^2 value in the fit output information
- [x] Fix naming of intermediate workspaces in case the optional FitOutputWorkspace property is not specified.

The loading feature along with the InputWorkspace is currently not added, as after fixing the renaming issue, the work so that it doesn't need to be pre-loaded each time.
   
The documentation for the PolarizedDiffractionILL has been updated with a more comprehensive example and explanation of the possibility to save time by caching the inputs.



**To test:**

1. Make sure you have ILL D7 UnitTest data in your user search directories
2. Load 402652_403041.nxs or other nxs file that contains underscore, e.g. using the example from [documentation](https://docs.mantidproject.org/nightly/algorithms/D7YIGPositionCalibration-v1.html)
3. Run the following script in mantid. You may need to modify the InputWorkspace and ApproximateWavelength relevant to your case:
```
from mantid.simpleapi import *

approximate_wavelength = '3.1' # Angstrom
D7YIGPositionCalibration(
		InputWorkspace='402652_403041',		
		ApproximateWavelength=approximate_wavelength,
                YIGPeaksFile='D7_YIG_peaks.xml',
                MinimalDistanceBetweenPeaks=1.5,
		BraggPeakWidth=1.5,
		BankOffsets=[0,0,0],
                MaskedBinsRange=[-50, -25, 15],
		FittingMethod='None',
		ClearCache=False,
                FitOutputWorkspace='fitting_test')
```

4. Verify that:
a) calibration runs without any issues,
b) there are no issues if you comment out the YIGPeaksFile property,
c) the input workspace is not be renamed to anything else,
d) if you change the FitOutputWorkspace property to empty string, the intermediate workspaces should contain 'calibration' string. 

Fixes #30861. 

*This does not require release notes* because these changes are small corrections that restore the expected behaviour.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
